.taskcluster.yml

```diff
@@ -101,199 +107,212 @@ tasks:
-                                  kind: 'action-callback'
+                                  kind: decision-task
                               else:
-                                  $if: 'tasks_for == "cron"'
+                                  $if: 'tasks_for == "action"'
                                   then:
-                                      kind: cron-task
-                      routes:
-                          $flatten:
-                              - checks
-                              - $if: 'level == "3"'
-                                then:
-                                    - "tc-treeherder.v2.${project}.${head_sha}"
-                                else: []
-                      scopes:
-                          # `https://` is 8 characters so, ${repoUrl[8:]} is the repository without the protocol.
-                          $if: 'tasks_for == "github-push"'
-                          then:
-                              $let:
-                                  short_head_branch:
-                                      $if: 'head_branch[:11] == "refs/heads/"'
-                                      then: {$eval: 'head_branch[11:]'}
-                                      else: ${head_branch}
-                              in:
-                                  - 'assume:repo:${repoUrl[8:]}:branch:${short_head_branch}'
-                          else:
-                              $if: 'tasks_for == "github-pull-request"'
+                                      kind: 'action-callback'
+                                  else:
+                                      $if: 'tasks_for == "cron"'
+                                      then:
+                                          kind: cron-task
+                          routes:
+                              $flattenDeep:
+                                  - checks
+                                  - $if: 'level == "3"'
+                                    then:
+                                        - tc-treeherder.v2.${project}.${head_sha}
+                                        # TODO Bug 1601928: Make this scope fork-friendly once ${project} is better defined. This will enable
+                                        # staging release promotion on forks.
+                                        - $if: 'tasks_for == "github-push"'
+                                          then:
+                                              - index.project.mobile.${project}.v2.branch.${short_head_branch}.latest.taskgraph.decision
+                                              - index.project.mobile.${project}.v2.branch.${short_head_branch}.revision.${head_sha}.taskgraph.decision
+                                        - $if: 'tasks_for == "cron"'
+                                          then:
+                                              - index.project.mobile.${project}.v2.branch.${short_head_branch}.latest.taskgraph.decision-${cron.job_name}
```

Comment:
@rail FYI: This route is slightly different than the [one Shipit uses](https://github.com/mozilla-releng/shipit/blob/b82699eac593b5b267f8036e19e5001014315f45/api/src/shipit_api/tasks.py#L39). See `branch` between `v2` and `${short_head_branch}`. 2 reasons to use this namespace:

1. Branches in hg aren't used the same way they are in git. In hg `${project}` is roughly equivalent to a branch. Eg: `mozilla-central` and `mozilla-beta`. 
2. https://firefox-ci-tc.services.mozilla.com/tasks/index/project.mobile.fenix.v2 is already listing build types. Mixing build types and branch names will confuse many people